[Stdlib][Overlays] Rename various classes to avoid conflicting ObjC names.

mikeash opened the pull request:
Old Swift and new Swift runtimes and overlays need to coexist in the same process. This means there must not be any classes which have the same ObjC runtime name in old and new, because the ObjC runtime doesn't like name collisions.

When possible without breaking source compatibility, classes were renamed in Swift, which results in a different ObjC name.

Public classes were renamed only on the ObjC side using the @_objcRuntimeName attribute.

This is similar to the work done in pull request #19295. That only renamed @objc classes. This renames all of the others, since even pure Swift classes still get an ObjC name.

rdar://problem/46646438

mikeash committed:
[Stdlib][Overlays] Rename various classes to avoid conflicting ObjC names.

Old Swift and new Swift runtimes and overlays need to coexist in the same process. This means there must not be any classes which have the same ObjC runtime name in old and new, because the ObjC runtime doesn't like name collisions.

When possible without breaking source compatibility, classes were renamed in Swift, which results in a different ObjC name.

Public classes were renamed only on the ObjC side using the @_objcRuntimeName attribute.

This is similar to the work done in pull request #19295. That only renamed @objc classes. This renames all of the others, since even pure Swift classes still get an ObjC name.

rdar://problem/46646438

mikeash commented:
@swift-ci please test

mikeash committed:
[AST] Fix the printing of @_objcRuntimeName.

rdar://problem/46646438

mikeash commented:
@swift-ci please test

mikeash commented:
https://github.com/apple/swift-lldb/pull/1154

@swift-ci please test

mikeash commented:
@swift-ci please test

mikeash commented:
@swift-ci please test

mikeash commented:
Helps to push ALL of the changes made to resolve the merge....

mikeash commented:
Also helps to test with the other PR it needs, sigh.

mikeash commented:
https://github.com/apple/swift-lldb/pull/1154

@swift-ci please test

jrose-apple commented on lib/AST/Attr.cpp:
```diff
@@ -536,10 +536,10 @@ bool DeclAttribute::printImpl(ASTPrinter &Printer, const PrintOptions &Options,
 
   case DAK_ObjCRuntimeName: {
-    Printer.printAttrName("@objc");
+    Printer.printAttrName("@_objcRuntimeName");
```
(That can come separately.)

mikeash commented:
https://github.com/apple/swift-lldb/pull/1154

@swift-ci please test linux platform

mikeash commented:
apple/swift-lldb#1154

@swift-ci please test

jrose-apple commented on stdlib/public/core/DictionaryStorage.swift:
```diff
@@ -180,21 +186,21 @@ extension _EmptyDictionarySingleton: _NSDictionaryCore {
 }
 
 @usableFromInline
 final internal class _DictionaryStorage<Key: Hashable, Value>
```
This one didn't need to be renamed because it's new? (And the same for the Set one.)

mikeash replied:
Turns out yes, but I (correctly) missed it by accident, so that's bad.

mikeash committed:
[Stdlib] Put $ at the end of the Swift module's generic class ObjC names.

This avoids name conflicts between generic classes in the Swift modules of the old and new stdlibs when loaded into the same process.

mikeash commented:
apple/swift-lldb#1154

@swift-ci please test

mikeash committed:
[Stdlib] Fix botched merge conflict resolution in Data.swift.

mikeash commented:
apple/swift-lldb#1154

@swift-ci please test
